xbmc/addons/addoninfo/AddonInfo.cpp

```diff
@@ -177,9 +177,11 @@ bool CAddonInfo::ProvidesSeveralSubContents() const
   return contents > 0 ? true : false;
 }
 
-bool CAddonInfo::MeetsVersion(const AddonVersion& versionMin, const AddonVersion& version) const
+bool CAddonInfo::MeetsVersion(const AddonVersion& versionMin, const AddonVersion& version,
+                              const AddonVersion& versionMax) const
 {
-  return !(versionMin > m_version || version < m_minversion);
+  return !(versionMin > m_version || (versionMax.empty() && version < m_minversion) ||
```

Comment:
Using the negation of the overall condition makes this tremendous hard to read (and error-prone) as the number of subconditions increase. Can you write it the other way around?

Reply:
Okay, I rewrote this and made sure that `versionMax` is never empty by using the `version` value:
https://github.com/xbmc/xbmc/blob/3fb7ad61c9f6451ba1bc0283d013230bafb2c2f6/xbmc/addons/addoninfo/AddonInfo.h#L39

So the overall condition could be simplified to:
https://github.com/xbmc/xbmc/blob/3fb7ad61c9f6451ba1bc0283d013230bafb2c2f6/xbmc/addons/addoninfo/AddonInfo.cpp#L183